### PR TITLE
Add endpoint for removing empty tags

### PR DESCRIPTION
**Pull request by tijmenb**
Tags without any documents tagged to it can be safely deleted. This endpoint will be used by collections-publisher to allow deletions of topics (and maybe browse pages in the future).

See https://github.com/alphagov/collections-publisher/pull/138 for more info. Trello: https://trello.com/c/ybdnGMhl

This is probably best reviewed by @alext or @rboulton to make sure this doesn't do anything unexpected.

### CODE REVIEW

**Comment by jackscotti** [comment]
:+1:

File: app/controllers/tags_controller.rb [hunk 1]
```diff
@@ -96,6 +96,19 @@ def publish
     end
   end
 
+  def destroy
+    respond_to do |format|
+      format.json do
+        if Artefact.with_tags([@tag.tag_id]).any?
+          render json: { error: 'Tag has documents tagged to it' }, status: :bad_request
```
Comment (alext) [inline app/controllers/tags_controller.rb]: I think a 409 (Conflict) would be a better status to return here.

File: app/controllers/tags_controller.rb [hunk 2]
```diff
@@ -96,6 +96,19 @@ def publish
...
+        else
+          @tag.destroy
+          render json: @tag, status: 204
```
Comment (alext) [inline app/controllers/tags_controller.rb]: 204 isn't appropriate here if we're returning the tag as JSON.  204 means "No Content"

**Comment by tijmenb** [comment]
Good points - updated.

**Comment by rboulton** [comment]
Still looks like it's returning a 204 status code.

**Comment by tijmenb** [comment]
Updated with a empty 200 for the deleted route.

**Pull request merged**